lib/google_contacts_api/group.rb

```diff
@@ -40,5 +40,18 @@ def edit_link
       _link = self["link"].find { |l| l.rel == "edit" }
       _link ? _link.href : nil
     end
+
+    def self.create(attrs, api)
+      response = api.post('groups/default/full', xml_for_create(attrs), {}, 'Content-Type' => 'application/atom+xml')
+      Group.new(Hashie::Mash.new(JSON.parse(response.body)).entry)
+    end
+
+    def self.xml_for_create(group_attrs)
+      @@new_group_template ||=
+        File.new(File.dirname(__FILE__) + '/templates/group.xml.erb').read
+
+      # Passes in group_attrs to the template via binding
+      ERB.new(@@new_group_template).result(binding)
```

Comment:
One thing I learned from doing group and contact creation in production is that there are some very small, subtle and annoying differences in what the XML should be depending on whether it's a contact create or update (and presumably that may apply to group create or update). For reference, here's the template code I'm using in my fork of the gem for the contact creation: https://github.com/CruGlobal/google_contacts_api/blob/master/lib/google_contacts_api/templates/contact.xml.erb.

The enclosing tag should be `atom:entry` for create and just `entry` for update. It's possible we could just use one or the other, but I do remember facing a confusing bug because the Google contacts API expected it in a specific format.

So while I definitely think that having a general XML generation strategy would be nice, I think that if we first approach it on a case by case basis using templates and we have good specs and a way to test things against the real Google contacts API, then if we can refactor in a clean way then we should do that eventually. But yeah, my thought is that templates are a good short/medium term approach.